### PR TITLE
Blog App: Validations and model specs

### PR DESCRIPTION
**Pull request by JuanDiegoGuerra**
### Activiy Requirements - Validations and model specs

- Installed RSpec-Rails library. ✔
- Created spec test file for User, Post, Like, and Comment models. ✔
- Add tests for validations and methods in User, Post, Comment, and Like models. ✔
- Tests are passing and everything is working as expected. ✔
image

### CODE REVIEW

File: spec/models/user_spec.rb
Comment (fobadara) [inline spec/models/user_spec.rb:1]: [REQUIRED] Kindly test for recent_comments and update_posts_counter methods. It is a requirement to test all custom methods.

File: spec/models/comment_spec.rb
Comment (fobadara) [inline spec/models/comment_spec.rb:1]: [REQUIRED] Kindly include tests for `update_post_comments_counter`. It is a requirement to test all custom methods.

File: spec/models/like_spec.rb
Comment (fobadara) [inline spec/models/like_spec.rb:1]: [REQUIRED] Kindly include tests for `update_post_likes_counter`. It is a requirement to test all custom methods.

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @JuanDiegoGuerra, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

REDO

## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code
✅ Good first attempt

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by JuanDiegoGuerra** [comment]
###  Required changes made 🛠

- **Fixed:** _"include tests for update_post_comments_counter"_ 🛠
- **Fixed:** _"include tests for update_post_likes_counter"_ 🛠
- Added some other Tests. There are now a total of 19 and all of the are passing! ✔
image

todolist1

**Review by algerina — APPROVED** [review]
Hi @JuanDiegoGuerra , 

Great job so far!  

May I highlight the following: 
- [x]  Well done adding the missing tests as required👏  ✔️ 
- [x] Your code is clean and well organized✔️



## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:

Your project is complete!  It's time to merge it :shipit:


 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

**Pull request merged**